openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -482,34 +566,9 @@ def _process_texture(self, filepath, do_maketx, staging, linearize, force):
                     ("Paths not found on disk, "
                      "skipping hardlink: %s") % (existing,)
                 )
-
-        if do_maketx and ext != ".tx":
-            # Produce .tx file in staging if source file is not .tx
-            converted = os.path.join(staging, "resources", fname + ".tx")
-
-            if linearize:
-                self.log.info("tx: converting sRGB -> linear")
-                colorconvert = "--colorconvert sRGB linear"
-            else:
-                colorconvert = ""
-
-            # Ensure folder exists
-            if not os.path.exists(os.path.dirname(converted)):
-                os.makedirs(os.path.dirname(converted))
-
-            self.log.info("Generating .tx file for %s .." % filepath)
-            maketx(
-                filepath,
-                converted,
-                # Include `source-hash` as string metadata
-                "-sattrib",
-                "sourceHash",
-                escape_space(texture_hash),
-                colorconvert,
-            )
-
-            return converted, COPY, texture_hash
-
+        for processor in processors:
+            processor().process(filepath)
```

Comment:
Should we make the `process` method a `staticmethod` or `classmethod` if it doesn't actually store anything on the Processor instance itself nor takes any init arguments?

Reply:
Also shouldn't we be using the returned filepath?
For example:
```processed_path = processor.process(filepath)```

Reply:
You are indeed correct, fixed!